src/_C014.ino

```diff
@@ -0,0 +1,686 @@
+                            CPlugin_014_sendMQTTnode(nodename, deviceName.c_str(), ExtraTaskSettings.TaskDeviceValueNames[varNr], "/$unit", unitName.c_str(), errorCounter);
+                          }
+                          unitName = "";
+                        }
+                      }
+                    } // end loop throug values
+                  } else { // Device has custom Values
+                    log = F("C014 : Device has custom values: ");
+                    log += getPluginNameFromDeviceIndex(Settings.TaskDeviceNumber[x]);
+                    addLog(LOG_LEVEL_INFO, log+" ?!")
+                  }
+                }
+                if (valuesList!="")
+                {
+                  // $properties	Device → Controller	Properties the node exposes, with format id separated by a , if there are multiple nodes.	Yes	Yes
+                  CPlugin_014_sendMQTTnode(nodename, deviceName.c_str(), "$properties", "", valuesList.c_str(), errorCounter);
+                  valuesList="";
+                }
+              } else { // device not enabeled
+                log = F("C014 : Device Disabled: ");
+                log += getPluginNameFromDeviceIndex(Settings.TaskDeviceNumber[x]);
+                addLog(LOG_LEVEL_INFO, log+" not propagated!")
+              }
+            } // device configured
+          } // loop through devices
+
+          // and finally ...
+          // $nodes	Device → Controller	Nodes the device exposes, with format id separated by a , if there are multiple nodes. To make a node an array, append [] to the ID.	Yes	Yes
+          CPlugin_014_sendMQTTdevice(pubname, "$nodes", nodesList.c_str(), errorCounter);
+        }
+
+        log = F("C014 : autodiscover information of ");
+        log += deviceCount;
+        log += F(" Devices and ");
+        log += nodeCount;
+        log += F(" Nodes sent with ");
+        if (errorCounter>0)
+        {
+          // alert: this is the state the device is when connected to the MQTT broker, but something wrong is happening. E.g. a sensor is not providing data and needs human intervention. You have to send this message when something is wrong.
+          CPlugin_014_sendMQTTdevice(pubname,"$state","alert",errorCounter);
+          log+=errorCounter;
+        } else {
+          // ready: this is the state the device is in when it is connected to the MQTT broker, has sent all Homie messages and is ready to operate. You have to send this message after all other announcements message have been sent.
+          CPlugin_014_sendMQTTdevice(pubname,"$state","ready",errorCounter);
+          log+=F("no");
+        }
+        log+=F(" errors! (");
+        log+=msgCounter;
+        log+=F(" messages)");
+        msgCounter = 0;
+        errorCounter = 0;
+        addLog(LOG_LEVEL_DEBUG, log);
```

Comment:
Try to wrap the log string creation in check for `logLevelActiveFor`

Reply:
done